homeassistant/components/vizio/config_flow.py

```diff
@@ -0,0 +1,169 @@
+                default=input_dict.get(CONF_VOLUME_STEP, DEFAULT_VOLUME_STEP),
+            ): vol.All(vol.Coerce(int), vol.Range(min=1, max=10)),
+        }
+    )
+
+
+class VizioConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Vizio config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self) -> config_entries.ConfigFlow:
+        """Initialize config flow."""
+        self.import_schema = None
+        self.user_schema = None
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> config_entries.OptionsFlow:
+        """Get the options flow for this handler."""
+        return VizioOptionsConfigFlow(config_entry)
+
+    async def async_step_user(
+        self, user_input: Dict[str, Any] = None
+    ) -> Dict[str, Any]:
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            # Store current values in case setup fails and user needs to edit
+            self.user_schema = update_schema_defaults(user_input)
+
+            # Check if new config entry matches any existing config entries
+            for entry in self.hass.config_entries.async_entries(DOMAIN):
+                if entry.data[CONF_HOST] == user_input[CONF_HOST]:
+                    errors[CONF_HOST] = "host_exists"
+                    break
+
+                if entry.data[CONF_NAME] == user_input[CONF_NAME]:
+                    errors[CONF_NAME] = "name_exists"
+                    break
+
+            if not errors:
+                try:
+                    # Ensure schema passes custom validation, otherwise catch exception and add error
+                    validate_auth(user_input)
+
+                    # Ensure config is valid for a device
+                    if not await VizioAsync.validate_config(
+                        user_input[CONF_HOST],
+                        user_input.get(CONF_ACCESS_TOKEN),
+                        user_input[CONF_DEVICE_CLASS],
+                    ):
+                        errors["base"] = "invalid_setup"
+                except vol.Invalid:
+                    errors["base"] = "tv_needs_token"
+
+            if not errors:
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME], data=user_input
+                )
+
+        schema = self.user_schema or self.import_schema or vol.Schema(VIZIO_SCHEMA)
+
+        return self.async_show_form(step_id="user", data_schema=schema, errors=errors)
+
+    async def async_step_import(self, import_config: Dict[str, Any]) -> Dict[str, Any]:
+        """Import a config entry from configuration.yaml."""
+        # Check if new config entry matches any existing config entries
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            if entry.data[CONF_HOST] == import_config[CONF_HOST] and entry.data[
```

Comment:
will this work even if I use a serial number as the unique ID after initialization?

Reply:
nevermind I see that this `unique_id` is for the config entry not the entity, so I need to set the `unique_id` on successful creation correct?

Reply:
You can set unique id earlier since the config entry won't be saved if its flow is aborted.

Reply:
I can set `unique_id` to HOST to start and use that as a quicker way to check, but since `unique_id` eventually gets set to the serial number, which I can only get after successful connection to the device, I still need to include this check

Reply:
Using serial for unique id is the most future proof. But as long as there is no discovery flow that can update a host address, the user will need to remove the entry and add a new entry to change host. So in that way, host is enough as unique id.

But since we have serial available, I suggest we go with that.

Reply:
OK, then I can't use `async_set_unique_id` because at that time I don't know the serial number

Reply:
Yes.